app/assets/javascripts/controllers/middleware_datasource/middleware_datasource_service.js

```diff
@@ -124,21 +124,24 @@ function MwAddDatasourceService($http, $q) {
     var deferred = $q.defer();
     var BASE_URL = '/middleware_server/jdbc_drivers';
     var parameterizedUrl = BASE_URL + '?server_id=' + serverId;
+    var serverError = 'internal_server_error';
 
     $http.get(parameterizedUrl).then(function(driverData) {
-      var transformedData = _.chain(driverData.data.data)
-        .map(function(driver) {
-          return {'id': driver.properties['Driver Name'].toUpperCase(),
-                  'label': driver.properties['Driver Name'],
-                  'moduleName': driver.properties['Module Name'],
-                  'xaDsClass': driver.properties['XA DS Class'],
-                  'driverClass': driver.properties['Driver Class']};
-      })
-      .value();
-
-      deferred.resolve(transformedData);
-    }).catch(function(errorMsg) {
-      deferred.reject(errorMsg);
+      if (driverData.data.status === serverError) {
```

Comment:
@karelhala I was wondering if you could use our standard pattern here for making API calls (that is currently being used in many other angular controllers)
```javascript
API.get(url)
  .then(getFunction)
  .catch(miqService.handleFailure);
```

Or in your case, it would be

```javascript
$http.get(url)
  .then(getFunction)
  .catch(miqService.handleFailure);
```

The advantage of making GET/POST calls like this is -
- consistency
- No need to deal with the error codes explicitly since `miqService.handleFailure` would do that for you.